presentation/src/main/java/com/example/healthysalad/OrderSummaryFragment.kt

```diff
@@ -35,27 +39,25 @@ class SelectBaseFragment : Fragment() {
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        //Data().loadItem("base")
-        // Inflate the layout for this fragment
-        var v : View = inflater.inflate(R.layout.fragment_select_base,container, false)
+        val view : View = inflater.inflate(R.layout.fragment_order_summary, container,false)
+        val viewManager: RecyclerView.LayoutManager = LinearLayoutManager(view.context)
+        val viewAdapter: OrderSummaryViewAdapter = OrderSummaryViewAdapter(Order)
+        val recyclerView : RecyclerView = view.findViewById(R.id.order_summary_recycler_view)
 
-        val openBackFrag: Button = v.findViewById(R.id.back_button)
-        openBackFrag.setOnClickListener {
-            activity!!.supportFragmentManager
-                .beginTransaction().replace(R.id.activity_main, TitleFragment())
-                .addToBackStack(null).commit()
-        }
-
-        val openNextFrag: Button = v.findViewById(R.id.next_button)
-        openNextFrag.setOnClickListener{
-            activity!!.supportFragmentManager
-                .beginTransaction().replace(R.id.activity_main, SelectProteinFragment())
-                .addToBackStack(null).commit()
-        }
+        recyclerView.setHasFixedSize(true)
+        recyclerView.layoutManager = viewManager
+        recyclerView.adapter = viewAdapter
 
 
-        return v
-
+        val openBackFrag: Button = view.findViewById(R.id.back_button)
+        openBackFrag.setOnClickListener {
+                activity!!.supportFragmentManager
```

Comment:
We could replace `activity!!` with `activity.let` function in case the instance is null. It's better to avoid `!!` usages unless we are certain the value is not null, since this can lead to NullPointerExceptions